File: Applications/CourtCounter/app/src/main/java/com/example/courtcounter/MainActivity.java
package com.example.courtcounter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    int scoreTeamA = 0;
    int scoreTeamB = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        displayForTeamA(0);
        displayForTeamB(0);
    }
    public void displayForTeamA(int score) {
        TextView scoreView = (TextView) findViewById(R.id.team_a_score);
        scoreView.setText(String.valueOf(score));
    }

    /**
     * Displays the given score for Team B.
     */
    public void displayForTeamB(int score) {
        TextView scoreView = (TextView) findViewById(R.id.team_b_score);
        scoreView.setText(String.valueOf(score));
    }

    public void addThreeButtonA(View view) {
        displayForTeamA( scoreTeamA = scoreTeamA + 3);

    }
    public void addTwoButtonA(View view) {
        displayForTeamA( scoreTeamA = scoreTeamA + 2);

    }
    public void addOneButtonA(View view) {
        displayForTeamA( scoreTeamA = scoreTeamA + 1);

    }
    public void addThreeButtonB(View view) {
        displayForTeamB( scoreTeamB = scoreTeamB + 3);

    }
    public void addTwoButtonB(View view) {
        displayForTeamB( scoreTeamB = scoreTeamB + 2);

    }
    public void addOneButtonB(View view) {
        displayForTeamB( scoreTeamB = scoreTeamB + 1);

    }

    public void resetButton(View view){
        displayForTeamA(0);
        displayForTeamB(0);
        scoreTeamB = 0;
        scoreTeamA = 0;
    }
}

File: Applications/ListViewDemo/app/src/main/java/com/example/listviewdemo/MainActivity.kt
package com.example.listviewdemo

import android.content.Context
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ListView
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val listView = findViewById<ListView>(R.id.main_listView)
       /* val redColor = Color.parseColor("#FF0000")
        listView.setBackgroundColor(redColor)*/

        listView.adapter =  MyCustomAdapter(this) // custom adapter for list
    }

    private class MyCustomAdapter(context:Context): BaseAdapter() {
        private val mContext: Context

        init {
            this.mContext = context
        }

        //responsible for how many rows in my list
        override fun getCount(): Int {
            return 5
        }

        override fun getItemId(position: Int): Long {
            return position.toLong()
        }
        override fun getItem(position: Int): Any {
            return "Test String"

        }

        //responsible for rending out each row
        override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
            val textView = TextView(mContext)
            textView.text = "HERE is my ROW for my LISTVIEW"
            return textView
        }
    }
}
